### PR TITLE
[Phase 2 - (Round 3) Method Reference]Lps 181008

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-181008

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:frontend

**Comment by Kvale1733** [comment]
@slnn No regressions.

File: modules/apps/frontend-js/frontend-js-top-head-extender/src/main/java/com/liferay/frontend/js/top/head/extender/internal/servlet/taglib/TopHeadDynamicInclude.java [hunk 1]
```diff
@@ -160,63 +212,79 @@ private void _addPortalBundles(List<String> urls, String propsKey) {
...
 
-		_addPortalBundles(_jsResourceURLs, PropsKeys.JAVASCRIPT_BAREBONE_FILES);
+	private ResourceURLsBag _rebuild() {
+		ResourceURLsBag resourceURLsBag = new ResourceURLsBag();
 
-		synchronized (_topHeadResourcesServiceReferences) {
-			for (ServiceReference<TopHeadResources>
-					topHeadResourcesServiceReference :
-						_topHeadResourcesServiceReferences) {
+		resourceURLsBag._allJsResourceURLs.clear();
```
Comment (dantewang) [inline modules/apps/frontend-js/frontend-js-top-head-extender/src/main/java/com/liferay/frontend/js/top/head/extender/internal/servlet/taglib/TopHeadDynamicInclude.java:236]: @slnn This is just created, why clear the list?

Comment (slnn) [reply modules/apps/frontend-js/frontend-js-top-head-extender/src/main/java/com/liferay/frontend/js/top/head/extender/internal/servlet/taglib/TopHeadDynamicInclude.java:236]: The class did it https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-top-head-extender/src/main/java/com/liferay/frontend/js/top/head/extender/internal/servlet/taglib/TopHeadDynamicInclude.java#L170, I just keep same logic as before.

Comment (dantewang) [reply modules/apps/frontend-js/frontend-js-top-head-extender/src/main/java/com/liferay/frontend/js/top/head/extender/internal/servlet/taglib/TopHeadDynamicInclude.java:236]: @slnn But now you re-create a new `Bag`, so the list in the new `Bag` is already empty.

**Pull request closed without merging**